pkg/preparer/orchestrate_test.go

```diff
@@ -248,27 +274,53 @@ func TestPreparerWillRequireSignatureWithKeyring(t *testing.T) {
 }
 
 func TestPreparerWillAcceptSignatureFromKeyring(t *testing.T) {
-	manifestBytes, err := testManifest(t).Bytes()
-	Assert(t).IsNil(err, "manifest bytes error should have been nil")
+	manifest, fakeSigner := testSignedManifest(t, nil)
 
-	fakeSigner, err := openpgp.NewEntity("p2", "p2-test", "p2@squareup.com", nil)
-	Assert(t).IsNil(err, "NewEntity error should have been nil")
+	p, _, fakePodRoot := testPreparer(t, &FakeStore{})
+	defer os.RemoveAll(fakePodRoot)
+	p.keyring = openpgp.EntityList{fakeSigner}
+
+	Assert(t).IsTrue(p.verifySignature(*manifest, logging.DefaultLogger), "should have accepted signed manifest")
+}
+
+func TestPreparerWillRejectSignatureForPreparerWithoutAuthorizedDeployers(t *testing.T) {
+	manifest, fakeSigner := testSignedManifest(t, func(m *pods.PodManifest, _ *openpgp.Entity) {
+		m.Id = "p2-preparer"
+	})
 
 	p, _, fakePodRoot := testPreparer(t, &FakeStore{})
 	defer os.RemoveAll(fakePodRoot)
 	p.keyring = openpgp.EntityList{fakeSigner}
 
-	var buf bytes.Buffer
-	sigWriter, err := clearsign.Encode(&buf, fakeSigner.PrivateKey, nil)
-	Assert(t).IsNil(err, "clearsign Encode error should have been nil")
+	Assert(t).IsFalse(p.verifySignature(*manifest, logging.DefaultLogger), "expected preparer to reject manifest (no authorized deployer)")
+}
 
-	sigWriter.Write(manifestBytes)
-	sigWriter.Close()
+func TestPreparerWillRejectUnauthorizedSignatureForPreparer(t *testing.T) {
+	manifest, fakeSigner := testSignedManifest(t, func(m *pods.PodManifest, _ *openpgp.Entity) {
+		m.Id = "p2-preparer"
+	})
 
-	manifest, err := pods.PodManifestFromBytes(buf.Bytes())
-	Assert(t).IsNil(err, "should have generated manifest from signed bytes")
+	p, _, fakePodRoot := testPreparer(t, &FakeStore{})
+	defer os.RemoveAll(fakePodRoot)
+	p.keyring = openpgp.EntityList{fakeSigner}
+	p.authorizedDeployers = []string{"nobodylol"}
 
-	Assert(t).IsTrue(p.verifySignature(*manifest, logging.DefaultLogger), "should have accepted signed manifest")
+	Assert(t).IsFalse(p.verifySignature(*manifest, logging.DefaultLogger), "expected preparer to reject manifest (unauthorized deployer)")
+}
+
+func TestPreparerWillAcceptAuthorizedSignatureForPreparer(t *testing.T) {
+	sig := ""
+	manifest, fakeSigner := testSignedManifest(t, func(m *pods.PodManifest, e *openpgp.Entity) {
+		m.Id = "p2-preparer"
```

Comment:
Can we make a `pods.PreparerID` instead of referring to this in places? That way when we generalize the concept (when we have time) it'll be easy to go through and remove references.